src/resmom/start_exec.c

```diff
@@ -3577,7 +3571,7 @@ start_process(task *ptask, char **argv, char **envp, bool nodemux)
 		case 0:	/* explicit reject */
 			free(progname);
 			free_attrlist(&argv_list);
-			free(env_str);
+			free_str_array(hook_output.env);
 			starter_return(kid_write, kid_read,
 					JOB_EXEC_FAILHOOK_DELETE, &sjr);
```

Comment:
shouldn't we be breaking out of this case? I see that the memory we just free in this case gets referenced again in the case below.